girder/constants.py

```diff
@@ -128,6 +128,10 @@ class SettingKey:
     EMAIL_HOST = 'core.email_host'
     REGISTRATION_POLICY = 'core.registration_policy'
     SMTP_HOST = 'core.smtp_host'
+    SMTP_PORT = 'core.smtp.port'
```

Comment:
It will work fine. ``None`` will get passed through as the port, and the same [internal logic](https://github.com/python/cpython/blob/2.7/Lib/smtplib.py#L304) will take over and split the host on ``:`` to get the port.